src/main/java/gregtech/common/mui/widget/workbench/CraftingInputSlot.java

```diff
@@ -68,10 +73,11 @@ public void draw(GuiContext context, WidgetTheme widgetTheme) {
         ItemStack itemstack = this.syncHandler.getStack();
         if (itemstack.isEmpty()) return;
 
-        guiScreen.setZ(100f);
-        guiScreen.getItemRenderer().zLevel = 100.0F;
-
         RenderUtil.renderItemInGUI(itemstack, 1, 1);
+        int allTints = logic.getTintLocations();
+        if ((allTints & 1 << this.syncHandler.index) != 0) {
+            RenderUtil.renderRect(0, 0, 18, 18, 100, 0x80FF0000);
+        }
 
         guiScreen.getItemRenderer().zLevel = 0.0F;
         guiScreen.setZ(0f);
```

Comment:
is there a reason you removed the z level adjustments? it causes the rect to render in front of certain items (notable GT tools)

Reply:
This is just an aesthetic choice for me (and perhaps makes it a bit easier to notice).